src/engine/platform/segapcm.cpp

```diff
@@ -159,7 +210,9 @@ int DivPlatformSegaPCM::dispatch(DivCommand c) {
     case DIV_CMD_NOTE_ON: {
       DivInstrument* ins=parent->getIns(chan[c.chan].ins,DIV_INS_AMIGA);
       if (skipRegisterWrites) break;
-      if (ins->type==DIV_INS_AMIGA) {
+      if (ins->type==DIV_INS_AMIGA || ins->type==DIV_INS_SEGAPCM) {
+        chan[c.chan].macroVolMul=(ins->type==DIV_INS_AMIGA)?64:127;
+        chan[c.chan].isNewSegaPCM=(ins->type==DIV_INS_SEGAPCM || parent->song.newSegaPCM);
```

Comment:
You just broke compatibility with songs that use Sample, have newSegaPCM flag on and use macros.

Reply:
Pushed a fix. Don't worry.